R/arc-select.R

```diff
@@ -4,8 +4,17 @@
 #'
 #' @param x an object of class `FeatureLayer` or `Table`.
 #' @param fields a character vector of the field names that you wish to be returned. By default all fields are returned.
+#' @param where a simple SQL where statement indicating which features should be selected.
 #' @param crs the spatial reference to be returned. If the CRS is different than the `FeatureLayer`'s CRS, a transformation will occur server-side. Ignored for `Table` objects.
-#' @param filter_geom an object of class `sfc` or `sfg` used to filter query results based on a predicate function. If an `sfc` object is provided it will be transformed to the layers spatial reference. If the `sfc` is missing a CRS (or is an `sfg` object) it is assumed to be in the layers spatial reference.
+#' @param filter_geom an object of class `bbox`, `sfc` or `sfg` used to filter
+#'   query results based on a predicate function. If an `sfc` object is provided
+#'   it will be transformed to the layers spatial reference. If the `sfc` is
+#'   missing a CRS (or is an `sfg` object) it is assumed to be in the layers
+#'   spatial reference. If an `sfc` object has multiple features, the features
+#'   are unioned with [sf::st_union()]. If an `sfc` object has MULTIPOLYGON
+#'   geometry, the features are treated as polygonal coverage and unioned with
+#'   `is_coverage = TRUE` before being cast to POLYGON geometry with
```

Comment:
One minor last-ish thing: can we move this detailed description to the @details?

Reply:
Sure. Mind if I clean up excess line lengths for the rest of the documentation in this script while I'm at it? Or do you prefer the single line for each argument definition as a style?

I can also run styler on the code if you use it but didn't want to clutter the PR with a bunch of minor spacing/indent changes if you had some non-tidyverse standard formatting you wanted to stick with.

Reply:
Let's try to keep the scope of the PR small.After this I'd like to add GHA for linting and R CMD check